core/templates/pages/exploration-editor-page/services/exploration-rights-backend-api.service.ts

```diff
@@ -0,0 +1,262 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A data service that stores data
+ * about the rights for this exploration.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import constants from 'assets/constants';
+import { ExplorationDataService } from
+  'pages/exploration-editor-page/services/exploration-data.service';
+import { AlertsService } from 'services/alerts.service';
+
+export interface ExplorationRightsBackendData {
+  rights: {
+    'editor_names': string[];
+    'owner_names': string[];
+    'voice_artist_names': string[];
+    'viewer_names': string[];
+    'status': string;
+    'cloned_from': string;
+    'community_owned': boolean;
+    'viewable_if_private': boolean;
+  };
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationRightsService {
+  ownerNames: string[] = undefined;
+  editorNames: string[] = undefined;
+  voiceArtistNames: string[] = undefined;
+  viewerNames: string[] = undefined;
+  _status: string = undefined;
+  _clonedFrom: string = undefined;
+  _isCommunityOwned: boolean = undefined;
+  _viewableIfPrivate: boolean = undefined;
+
+  constructor(
+    private alertsService: AlertsService,
+    private httpClient: HttpClient,
+    private explorationDataService: ExplorationDataService,
+  ) { }
+
+  init(
+      ownerNames: string[], editorNames: string[], voiceArtistNames: string[],
+      viewerNames: string[], status: string, clonedFrom: string,
+      isCommunityOwned: boolean, viewableIfPrivate: boolean): void {
+    this.ownerNames = ownerNames;
+    this.editorNames = editorNames;
+    this.voiceArtistNames = voiceArtistNames;
+    this.viewerNames = viewerNames;
+    this._status = status;
+    this._clonedFrom = clonedFrom;
+    this._isCommunityOwned = isCommunityOwned;
+    this._viewableIfPrivate = viewableIfPrivate;
+  }
+
+  clonedFrom(): string {
+    return this._clonedFrom;
+  }
+
+  isCloned(): boolean {
+    return Boolean(this._clonedFrom);
+  }
+
+  isPublic(): boolean {
+    return this._status === constants.ACTIVITY_STATUS_PUBLIC;
+  }
+
+  isPrivate(): boolean {
+    return this._status === constants.ACTIVITY_STATUS_PRIVATE;
+  }
+
+  viewableIfPrivate(): boolean {
+    return this._viewableIfPrivate;
+  }
+
+  isCommunityOwned(): boolean {
+    return this._isCommunityOwned;
+  }
+
+  makeCommunityOwned(): Promise<void> {
+    let requestUrl = (
```

Comment:
Please try to use const for URLs which are not changing.

Reply:
Done